### PR TITLE
Bugfix/auto cd revert

### PR DESCRIPTION
**Pull request by Shushman**
This addresses #44

### CODE REVIEW

File: src/TikzPictures.jl [hunk 1]
```diff
@@ -202,24 +202,9 @@ function _mktempdir(foldername)
...
-    if isempty(foldername)
-        foldername = "."
-    end
-
-    original_dir = abspath(".")
-    working_dir = abspath(foldername) # May be equal to original_dir
 
-    # Switch to working directory
-    cd(working_dir)
+function savePDFTPHelper(basefilename::AbstractString, foldername::AbstractString, tp::TikzPicture)
```
Comment (mykelk) [inline src/TikzPictures.jl]: You can probably delete the `TP` part of the name, right?

File: src/TikzPictures.jl [hunk 2]
```diff
@@ -330,25 +318,28 @@ function save(f::PDF, td::TikzDocument)
...
     working_dir = abspath(foldername) # May be equal to original_dir
 
-    # Switch to working directory
-    cd(working_dir)
+    # Call anonymous function to do task and automatically return
+    cd(() -> savePDFTDHelper(basefilename,foldername,td), working_dir)
+end
+
+
+function saveSVGTPHelper(basefilename::AbstractString, foldername::AbstractString, tp::TikzPicture, working_dir::AbstractString)
```
Comment (mykelk) [inline src/TikzPictures.jl]: no need for TP (also helper can probably be lowercase)

**Comment by zsunberg** [comment]
I ran the new branch with this script: https://github.com/JuliaPOMDP/POMDPGallery.jl/blob/v1/problems/LaserTag/script.jl and got the following error:

```
ERROR: LoadError: UndefVarError: f not defined
Stacktrace:
 [1] savePDFTPHelper(::String, ::String, ::TikzPictures.TikzPicture) at /home/zach/.julia/packages/TikzPictures/2XDDW/src/TikzPictures.jl:259
```

You could include that script in the tests, but it has a lot of esoteric dependencies on POMDPs stuff. Ideally all the code branches should be tested, but it is really hard to make tests for all :/

File: src/TikzPictures.jl (hunk 1 above)
Comment (mykelk) [reply src/TikzPictures.jl]: Also, in this function you have `save(f, tp)` but `f` is not defined.

**Comment by Shushman** [comment]
Yeah I'm all for cleaner do blocks, and that `f` was a silly miss. How about now? @zsunberg

**Comment by zsunberg** [comment]
Yep, it works perfectly now! Thanks!

**Pull request merged**